### PR TITLE
Compose styles with arrays in sx prop

### PR DESCRIPTION
**Pull request by jonavila**
This PR attempts to revive the work done @phelma (https://github.com/phelma) in #174. I took his work and put in the core package and update the `SxStyleProp` type. 

Sorry about the noise added by Prettier in the test file.

### CODE REVIEW

File: packages/core/src/index.ts [hunk 1]
```diff
@@ -16,9 +16,10 @@ export * from './types'
 const getCSS = props => {
   if (!props.sx && !props.css) return undefined
   return theme => {
-    const styles = css(props.sx)(theme)
+    const sx = Array.isArray(props.sx) ? props.sx : [props.sx]
+    const styles = sx.map(s => css(s)(theme))
```
Comment (jonavila) [inline packages/core/src/index.ts]: @jxnblk Do you have a preferred method for recursively flattening arrays? I realized after I opened the PR that Emotion recursively flattens arrays of styles and this doesn't account for that scenario.

Comment (lachlanjc) [reply packages/core/src/index.ts]: There are merge utilities in this file (see line 66)—is that what you’re looking for?

Comment (jonavila) [reply packages/core/src/index.ts]: @lachlanjc Thanks for the suggestion. Looks like the merge utility didn't have what i was looking for. I update a PR to do the flattening though.

**Comment by jorisw** [comment]
Please make this PR part of 0.4.0.

**Pull request closed without merging**